shared_model/builders/common_objects/common.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_BUILDERS_COMMON_HPP
+#define IROHA_BUILDERS_COMMON_HPP
+
+#include "common/result.hpp"
+#include "utils/polymorphic_wrapper.hpp"
+#include "validators/answer.hpp"
+
+// TODO: 16.02.2018 nickaleks: Add validators for common_objects IR-986
+
+namespace shared_model {
+  namespace builder {
+
+    /**
+     * BuilderResult represents return value of a builder,
+     * it either contains a value of ModelType, an error,
+     * which indicates why construction of an object went wrong.
+     *
+     * Since ModelType is an abstract class, its value has to be wrapped
+     * in something which has polymorphic behaviour, such as PolymorphicWrapper
```

Comment:
Sh_p is not a polywrapper. Better to remove this part of the documentation.

Reply:
Will fix